### DISCUSSION TITLE
WARNING: REMOTE HOST IDENTIFICATION HAS CHANGED!

### DISCUSSION THREAD

**Discussion by Lobana-sky** [community/community#29360] [Actions]
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@       WARNING: POSSIBLE DNS SPOOFING DETECTED!          @
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
The ECDSA host key for [github.com](http://github.com/) has changed,
and the key for the corresponding IP address 140.82.121.4
has a different value. This could either mean that
DNS SPOOFING is happening or the IP address for the host
and its host key have changed at the same time.
Offending key for IP in /home/.ssh/known_hosts:2
  remove with:
  ssh-keygen -f "/home/.ssh/known_hosts" -R "140.82.121.4"
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@    WARNING: REMOTE HOST IDENTIFICATION HAS CHANGED!     @
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
IT IS POSSIBLE THAT SOMEONE IS DOING SOMETHING NASTY!
Someone could be eavesdropping on you right now (man-in-the-middle attack)!
It is also possible that a host key has just been changed.
The fingerprint for the ECDSA key sent by the remote host is
SHA256:XXXXXXXXXXXXXXXXXXXXXXX.
Please contact your system administrator.
Add correct host key in /home/.ssh/known_hosts to get rid of this message.
Offending ECDSA key in /home/.ssh/known_hosts:3
  remove with:
  ssh-keygen -f "/home/.ssh/known_hosts" -R "[github.com](http://github.com/)"
ECDSA host key for [github.com](http://github.com/) has changed and you have requested strict checking.
Host key verification failed.
fatal: Could not read from remote repository.
Please make sure you have the correct access rights
and the repository exists.

**Comment by airtower-luna** [comment]
You censored the important part, which is also and not at all secret:

> The fingerprint for the ECDSA key sent by the remote host is
> SHA256:XXXXXXXXXXXXXXXXXXXXXXX.
> Please contact your system administrator.

You need to compare the fingerprint with the [published GitHub SSH public key fingerprints](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/githubs-ssh-key-fingerprints). If it is one of those you need to update your SSH configuration to recognize it. Otherwise, as the SSH message says:

> IT IS POSSIBLE THAT SOMEONE IS DOING SOMETHING NASTY!

"Nasty" referring to trying to intercept your SSH connection.

**Reply by Lobana-sky** [reply]
Hello @airtower-luna , thank you for your reply.

my fingerprint is not one of the [published GitHub SSH public key fingerprints](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/githubs-ssh-key-fingerprints), and I am sure it is not nasty attack.

that is happened whenever I create a new branch and try to push it, or pull from remote branch.
I tried ssh-keygen -R github.com, and delete known_host file, and other ways, but nothing work.

any other suggestion?

**Reply by airtower-luna** [reply]
> my fingerprint is not one of the [published GitHub SSH public key fingerprints](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/githubs-ssh-key-fingerprints), and I am sure it is not nasty attack.

The question is what the *remote host* fingerprint is. Your own key fingerprint should be different. :wink: 

Anyway, to say more I'd need to see debug output from SSH. Try:

```
ssh -vT git@github.com
```

And don't censor the remote host fingerprint.

**Reply by Lobana-sky** [reply]
@airtower-luna 

// ♥ ssh -vT git@github.com
OpenSSH_7.6p1 Ubuntu-4ubuntu0.5, OpenSSL 1.0.2n  7 Dec 2017
debug1: Reading configuration data /home/.ssh/config
debug1: /home/.ssh/config line 1: Applying options for github.com
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 19: Applying options for *
debug1: Connecting to github.com [140.82.121.3] port 22.
debug1: Connection established.
debug1: identity file /home/.ssh/id_rsa type 0
debug1: key_load_public: No such file or directory
debug1: identity file /home/.ssh/id_rsa-cert type -1
debug1: Local version string SSH-2.0-OpenSSH_7.6p1 Ubuntu-4ubuntu0.5
debug1: Remote protocol version 2.0, remote software version babeld-c0dd26fb
debug1: no match: babeld-c0dd26fb
debug1: Authenticating to github.com:22 as 'git'
debug1: SSH2_MSG_KEXINIT sent
debug1: SSH2_MSG_KEXINIT received
debug1: kex: algorithm: curve25519-sha256
debug1: kex: host key algorithm: ecdsa-sha2-nistp256
debug1: kex: server->client cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
debug1: kex: client->server cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
debug1: expecting SSH2_MSG_KEX_ECDH_REPLY
debug1: Server host key: ecdsa-sha2-nistp256 SHA256:9mnXIiLHmCALacN76iCzL8qzNBp1hBXPl2EBMRqMzcQ
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@    WARNING: REMOTE HOST IDENTIFICATION HAS CHANGED!     @
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
IT IS POSSIBLE THAT SOMEONE IS DOING SOMETHING NASTY!
Someone could be eavesdropping on you right now (man-in-the-middle attack)!
It is also possible that a host key has just been changed.
The fingerprint for the ECDSA key sent by the remote host is
SHA256:9mnXIiLHmCALacN76iCzL8qzNBp1hBXPl2EBMRqMzcQ.
Please contact your system administrator.
Add correct host key in /home/.ssh/known_hosts to get rid of this message.
Offending ECDSA key in /home/.ssh/known_hosts:1
  remove with:
  ssh-keygen -f "/home/.ssh/known_hosts" -R "github.com"
ECDSA host key for github.com has changed and you have requested strict checking.
Host key verification failed.

**Reply by airtower-luna** [reply]
Okay, that tells me two things:

1. The remote host fingerprint is *not* one of those GitHub publishes. That means danger.
2. The server IP is one listed for "web" by https://api.github.com/meta, so DNS seems fine.

So you're connecting to a correct IP address, and getting a wrong key. Which means *something* is messing with your SSH connection. Could be a lot of things:

* Misconfigured virus scanner or "firewall" trying to intercept your SSH connection for inspection
* Misconfigured router or firewall redirecting your SSH connection to some other server
* Attacker somewhere along the route
* Possibly other, vaguely similar thing

I think it's unlikely it's an error on the GitHub side, simply because it's been a few days since your first post and I haven't seen anyone else post about this kind of problem. :wink: Might be time to dig around your network, or if you're on a corporate/university/etc. network ask your admins.

Another peculiar thing is that (considering that you're posting here) your HTTPS connection seems to be fine. Would you mind sharing what certificate you see for `github.com` in your browser? :thinking:

**Comment by juanarbol** [comment]
I had the same issue. The solution for me was to connect to a VPN; not really sure if my ISP has some issues or something

**Reply by Lobana-sky** [reply]
also did not works for me.

**Reply by airtower-luna** [reply]
That's an interesting point, actually. Assuming the VPN isn't intercepted (you should check that, just like the HTTPS certificate I mentioned above) that'd mean the problem is on your machine, probably either misconfigured virus scanner or a virus. :sweat_smile:

**Comment by SuShuilian** [comment]
Here is the answer: https://github.blog/2023-03-23-we-updated-our-rsa-ssh-host-key/

**Comment by quantuminformation** [comment]
just saw this today

**Comment by h0jeZvgoxFepBQ2C** [comment]
Just now we have the same issue again?

The fingerprint for the ECDSA key sent by the remote host is
SHA256:Np4C3wlpg6klsffnRB5m/2nuPo3HW9QStKfhXOShv84.

The fingerprint for the ED25519 key sent by the remote host is
SHA256:Nv99lRmk4rW5Tinwyn7ljcBofffwgE/vmeeaj03QTsw.

? Github didnt post any key rotation?

**Reply by airtower-luna** [reply]
These are different from the posted ones at https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/githubs-ssh-key-fingerprints and I haven't seen any change either. So probably *something* is messing with your connection.
